### PR TITLE
Implemented the logic to support Nonce Filter on /api/v1/transactions/{hash}

### LINKED ISSUE #9715: Nonce Filter Is Ignored on /api/v1/transactions/{hash}
### Description

The nonce query param is only respected when making request with the transaction id and not when request is made with transaction hash.

### Steps to reproduce

Given the following request:
```bash
 curl -X 'GET' \                                                                                                                                                                                        
'https://testnet.mirrornode.hedera.com/api/v1/transactions/r6rpaXaYCvy9pPnCfO0f64IoDf0TrnNU0dHfFFbeOeBvhivdO+rcP134dCb44LkA?nonce=0' \
 -H 'accept: application/json'
```
The following result will be returned(correct):
```json
{"transactions":[{"bytes":null,"charged_tx_fee":207448,"consensus_timestamp":"1730913223.447111312","entity_id":null,"max_fee":"2655000000","memo_base64":"","name":"ETHEREUMTRANSACTION","nft_transfers":[],"node":"0.0.3","nonce":0,"parent_consensus_timestamp":null,"result":"WRONG_NONCE","scheduled":false,"staking_reward_transfers":[],"token_transfers":[],"transaction_hash":"r6rpaXaYCvy9pPnCfO0f64IoDf0TrnNU0dHfFFbeOeBvhivdO+rcP134dCb44LkA","transaction_id":"0.0.902-1730913210-989942481","transfers":[{"account":"0.0.3","amount":7413,"is_approval":false},{"account":"0.0.98","amount":180032,"is_approval":false},{"account":"0.0.800","amount":20003,"is_approval":false},{"account":"0.0.902","amount":-207448,"is_approval":false}],"valid_duration_seconds":"120","valid_start_timestamp":"1730913210.989942481"}]}
```
However with the following request with the wrong nonce filter still returns the result
```bash
curl -X 'GET' \                                                                                                              
'https://testnet.mirrornode.hedera.com/api/v1/transactions/r6rpaXaYCvy9pPnCfO0f64IoDf0TrnNU0dHfFFbeOeBvhivdO+rcP134dCb44LkA?nonce=1000' \
 -H 'accept: application/json'
```

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.117.0

### Operating system

None

### PR DESCRIPTION
**Pull request by HarshSawarkar**
**Description**:

Added the logic to filter based out of nonce.

**Related issue(s)**:

Fixes #9715 

**Notes for reviewer**:

Check for `if(isTransactionHash)` in `extractSqlFromTransactionsByIdOrHashRequest` in `transactions.js`

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/transactions.js [hunk 1]
```diff
@@ -738,6 +738,18 @@ const extractSqlFromTransactionsByIdOrHashRequest = async (transactionIdOrHash,
       .map((pos) => `$${pos}`)
       .join(',');
     mainConditions.push(`${Transaction.CONSENSUS_TIMESTAMP} in (${timestampPositions})`);
+    let nonce;
```
Comment (jnels124) [inline hedera-mirror-rest/transactions.js:741]: Can remove this block here and move the similar logic in the else block to be outside the if/else. It is ok to apply this logic to the scheduled parameter too.

File: hedera-mirror-rest/transactions.js
Comment (jnels124) [inline hedera-mirror-rest/transactions.js:1]: Needs test cases to test filtering by nonce and scheduled

- [ ] nonce and scheduled match
- [ ] nonce matches scheduled doesn't
- [ ] scheduled matches nonce doesn't
- [ ] neither match

Comment (jnels124) [reply hedera-mirror-rest/transactions.js:1]: Relevant tests to modify will be found in 
`hedera-mirror-rest/__tests__/specs/transactions/{id}/`

File: hedera-mirror-rest/transactions.js (hunk 1 above)
Comment (HarshSawarkar) [reply hedera-mirror-rest/transactions.js]: Updated

File: hedera-mirror-rest/transactions.js
Comment (HarshSawarkar) [reply hedera-mirror-rest/transactions.js:1]: Modified

Comment (jnels124) [reply hedera-mirror-rest/transactions.js:1]: Was going to apply these changes on your branch directly but seems pushing to your branch is not allowed. Can you please apply the changes in this [commit](https://github.com/hashgraph/hedera-mirror-node/commit/79892c9f06970eabfdad2868f9e7b445564cfa97)?

Comment (HarshSawarkar) [reply hedera-mirror-rest/transactions.js:1]: Updated

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Comment by HarshSawarkar** [comment]
> Thanks for your contribution!

Thank You.Looking forward to make more contributions.

**Pull request merged**